agent/lm_agent/tokenstat.py

```diff
@@ -259,32 +263,31 @@ async def report() -> typing.List[dict]:
     view of what features are available with what actually exists in the
     license server database.
     """
-    tool_awaitables = []
-    reconciliation = []
-    tools = ToolOptionsCollection.tools
+    report_items = []
 
-    # Iterate over the license servers and features appending to list
-    # of tools/cmds to be ran.
     license_configurations = await get_config_from_backend()
     local_licenses = get_all_product_features_from_cluster(await scontrol_show_lic())
     filtered_entries = get_local_license_configurations(license_configurations, local_licenses)
 
+    license_server_interface: LicenseServerInterface
+
     for entry in filtered_entries:
-        for license_server_type in tools:
-            if entry.license_server_type == license_server_type:
-                options = tools[license_server_type]
-                for feature in entry.features.keys():
-                    tool_awaitables.append(
-                        attempt_tool_checks(options, entry.product, feature, entry.license_servers)
-                    )
-    # run all checkers in parallel
-    results = await asyncio.gather(*tool_awaitables, return_exceptions=True)
-    for res in results:
-        if isinstance(res, Exception):
-            formatted = traceback.format_exception(type(res), res, res.__traceback__)
-            logger.error("".join(formatted))
+        product_features_to_check = []
+        for feature in entry.features.keys():
+            product_feature = f"{entry.product}.{feature}"
+            product_features_to_check.append(product_feature)
+
+        if entry.license_server_type == "flexlm":
+            license_server_interface = FlexLMLicenseServer(entry.license_servers)
+        elif entry.license_server_type == "rlm":
+            license_server_interface = RLMLicenseServer(entry.license_servers)
         else:
```

Comment:
As we add more license server types, it might be good to create a mapping (dict), attempt to fetch the appropriate type from it, and raise the exception if it cannot be found:

```suggestion
        server_type_map = dict(
            flexlm=FlexLMLicenseServer,
            rlm=RLMLicenseServer,
        )
        server_type = server_type_map.get(entry.license_server_type)
        if server_type is None:
            raise LicenseManagerNonSupportedServerTypeError()
        license_server_interface = server_type(entry.license_servers)
```

Reply:
Nice, I addressed in the latest commit. Now it looks much better!